apps/dashboard/test/config/configuration_singleton_test.rb

```diff
@@ -213,73 +212,86 @@ def no_config_env
   end
 
   test "can configure announcement path" do
-    ENV["OOD_ANNOUNCEMENT_PATH"] = "/path/to/announcement"
-    assert_equal Pathname.new("/path/to/announcement"), ConfigurationSingleton.new.announcement_path
+    with_modified_env(OOD_ANNOUNCEMENT_PATH: '/path/to/announcement') do
+      assert_equal Pathname.new('/path/to/announcement'), ConfigurationSingleton.new.announcement_path
+    end
   end
 
   test "should have default developer docs url" do
     assert_equal "https://go.osu.edu/ood-app-dev", ConfigurationSingleton.new.developer_docs_url
   end
 
   test "can configure developer docs url" do
-    ENV["OOD_DASHBOARD_DEV_DOCS_URL"] = "https://www.example.com"
-    assert_equal "https://www.example.com", ConfigurationSingleton.new.developer_docs_url
+    with_modified_env(OOD_DASHBOARD_DEV_DOCS_URL: 'https://www.example.com') do
+      assert_equal 'https://www.example.com', ConfigurationSingleton.new.developer_docs_url
+    end
   end
 
   test "should hide the all apps link by default" do
     refute ConfigurationSingleton.new.show_all_apps_link?
   end
 
   test "can enable the all apps link" do
-    ENV["SHOW_ALL_APPS_LINK"] = "true"
-    assert ConfigurationSingleton.new.show_all_apps_link?
+    with_modified_env(SHOW_ALL_APPS_LINK: 'true') do
+      assert ConfigurationSingleton.new.show_all_apps_link?
+    end
   end
 
   test "should have default dataroot under app if not production" do
     assert_equal Rails.root.join("data"), ConfigurationSingleton.new.dataroot
   end
 
   test "should have default dataroot under home dir if production" do
-    ENV["RAILS_ENV"] = "production"
-    assert_equal Pathname.new("~/ondemand/data/sys/dashboard").expand_path, ConfigurationSingleton.new.dataroot
+    with_modified_env(RAILS_ENV: 'production') do
+      ENV.delete('OOD_DATAROOT')
```

Comment:
I think setting it to `nil` here works the same instead of deleting. 

```suggestion
    with_modified_env(RAILS_ENV: 'production', OOD_DATAROOT: nil) do
```